src/EA.Weee.RequestHandlers/Admin/Reports/GetUkWeeeAtAatfsCsv/GetUkWeeeAtAatfsCsvDataAccess.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace EA.Weee.RequestHandlers.Admin.Reports.GetUkWeeeAtAatfsCsv
+{
+    using System.Collections.Generic;
+    using System.Data.Entity;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using EA.Weee.Core.AatfReturn;
+    using EA.Weee.DataAccess;
+
+    public class GetUkWeeeAtAatfsCsvDataAccess : IGetUkWeeeAtAatfsCsvDataAccess
+    {
+        private readonly WeeeContext context;
+
+        public GetUkWeeeAtAatfsCsvDataAccess(WeeeContext context)
+        {
+            this.context = context;
+        }
+
+        /// <summary>
+        /// Returns all data returns for the specified compliance year.
+        /// Each data return will be pre-loaded with the current version's
+        /// set of collected amounts and delivered amounts.
+        /// The results will not be ordered.
+        /// </summary>
+        public async Task<IEnumerable<PartialAatfReturn>> FetchPartialAatfReturnsForComplianceYearAsync(int complianceYear)
+        {
+            var parentIds = await context.Returns
```

Comment:
Is the checking of the parent Id's to try and calculate the latest submitted returns?

Reply:
Yes, it's essentially saying 'if an AATF Return is a parent, then ignore it'

Reply:
The parent Id is just to store which return a return was copied from, i.e. when the edit link is clicked
I think this query will probably be excluding the latest as each edit will have a parent, and each edit will be the most recent? (I think) Probably need to group by Organisation and take the top one ordered by Submitted date desc and where return status = 2

Reply:
I think it will still work, as the latest edit will not be a parent to any other report, so it shouldn't be excluded.
The way you suggest sounds more logical though, so I'll implement that - the code will be easier to understand too!

Reply:
Pushed changes and test updates